alertwildfire_live_recording.py

```diff
@@ -0,0 +1,162 @@
+fuegoRoot = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, os.path.join(fuegoRoot, 'lib'))
+sys.path.insert(0, fuegoRoot)
+
+import settings
+settings.fuegoRoot = fuegoRoot 
+import alertwildfire_API
+import goog_helper
+import collect_args
+import tempfile
+import time
+import shutil
+from multiprocessing import Pool
+import math
+import datetime
+import img_archive
+
+
+
+
+
+def cleanup_archive(googleServices, timethreshold):
+    """initializes a continual cleaning function that only acts to remove archived data past a given threshold.
+    Args:
+        googleServices: Drive service (from getGoogleServices())
+        timethreshold (flt): hours to keep data in archive
+    Returns:
+        continual should never return must be manually killed
+    """
+    img_archive.getImgPath
+    timestamp = time.mktime(datetime.datetime.now().timetuple()) -60*timethreshold
+    current_target = img_archive.getImgPath("./", "test", timestamp)[-23:-4]
+    while True:
+        for folder in goog_helper.driveListFilesByName(googleServices['drive'], settings.alertwildfire_archive):
+            for fileobj in goog_helper.driveListFilesByName(googleServices['drive'], folder['id']):
+                if fileobj['name'][-23:-4]< current_target:
+                    logging.error('deleting file', fileobj['name'])
+                    goog_helper.deleteItem(googleServices['drive'], file_id)
+            
+    return True
+
+def capture_and_record(googleServices, outputDir, cameras_in_drive, camera_name):
+    """requests current image from camera and uploads it to drive
+    Args:
+        googleServices: Drive service (from getGoogleServices())
+        outputDir (str): folder path to download into
+        cameras_in_drive (dict): dictionary of all camera archive folders with respective google IDs
+        camera_name (str): name of camera as recorded by alertwildfire
+    Returns:
+        imgPath: local path to downloaded object
+    """
+    if not camera_name in cameras_in_drive.keys():
+        cameras_in_drive[camera_name] = goog_helper.createFolder(googleServices['drive'], settings.alertwildfire_archive,  camera_name)
+    dirID = cameras_in_drive[camera_name]
+    imgPath = alertwildfire_API.request_current_image(outputDir, camera_name)
+    goog_helper.uploadFile(googleServices['drive'], dirID, imgPath)
+    print(imgPath)
+
+
+def camera_management(obj):
+    """manages the continual observation of a given set of cameras to watch.
+    Args:
+        obj (tuple): holds the googleServices, cameras_in_drive, camera_names_to_watch arguments
+            googleServices: Drive service (from getGoogleServices())
+            cameras_in_drive (dict): dictionary of all camera archive folders with respective google IDs
+            camera_names_to_watch (List): list of camera names that are to be watched by this process
+    Returns:
+        None
+    """
+    googleServices, cameras_in_drive, camera_names_to_watch = obj[0], obj[1], obj[2]
```

Comment:
the reason for this was only to allow for parallelization however since we stepped away from that i would like to return it to its original multi argument state